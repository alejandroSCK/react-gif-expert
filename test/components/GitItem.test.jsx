import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('GitItem unit test', () => {
    const title = "one punch man";
    const url = "https://media3.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=0628090eapy3zmtf83ya95eb6gy63upxucya2zh3po1myec2&rid=giphy.gif&ct=g";
    
    test('should match de snatshot', () => {
        const { container } = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('should show the image with the url and the alt', () => {
        render(<GifItem title={title} url={url}/>);
        const { src, alt } = screen.getByRole('img');
        expect( src).toBe(url);
        expect( alt).toBe(title);
    });

    test('should show the title', () => {
        render(<GifItem title={title} url={url}/>);
        expect( screen.getByText(title)).toBeTruthy();
    })
})