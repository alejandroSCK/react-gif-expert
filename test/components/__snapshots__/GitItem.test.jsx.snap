// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitItem unit test should match de snatshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="one punch man"
      src="https://media3.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=0628090eapy3zmtf83ya95eb6gy63upxucya2zh3po1myec2&rid=giphy.gif&ct=g"
    />
    <p>
      one punch man
    </p>
  </div>
</div>
`;
